lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosValidation.cxx

```diff
@@ -0,0 +1,184 @@
+/**
+ *  @brief Validation of a functional chaos expansion
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <algorithm>
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/FunctionalChaosValidation.hxx"
+#include "openturns/Os.hxx"
+#include "openturns/OSS.hxx"
+#include "openturns/DesignProxy.hxx"
+#include "openturns/LeastSquaresMethod.hxx"
+#include "openturns/KFoldSplitter.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(FunctionalChaosValidation)
+
+static const Factory<FunctionalChaosValidation> Factory_FunctionalChaosValidation;
+
+/* Default constructor */
+FunctionalChaosValidation::FunctionalChaosValidation()
+  : PersistentObject()
+{}
+
+/* Default constructor */
+FunctionalChaosValidation::FunctionalChaosValidation(const FunctionalChaosResult & functionalChaosResult)
+  : PersistentObject()
+  , functionalChaosResult_(functionalChaosResult)
+{
+  // Nothing to do
+}
+
+
+/* Virtual constructor */
+FunctionalChaosValidation * FunctionalChaosValidation::clone() const
+{
+  return new FunctionalChaosValidation(*this);
+}
+
+/* String converter */
+String FunctionalChaosValidation::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << FunctionalChaosValidation::GetClassName()
+      << " functional chaos result=" << functionalChaosResult_;
+  return oss;
+}
+
+/* Get result*/
+FunctionalChaosResult FunctionalChaosValidation::getResult() const
+{
+  return functionalChaosResult_;
+}
+
+/* Compute LOO error */
+Point FunctionalChaosValidation::computeLeaveOneOutError() const
+{
+  const FunctionCollection reducedBasis(functionalChaosResult_.getReducedBasis());
+  const Function transformation(functionalChaosResult_.getTransformation());
+  const Sample inputSample(functionalChaosResult_.getInputSample());
+  const Sample standardSample(transformation(inputSample));
+  DesignProxy designProxy(standardSample, reducedBasis);
+  const UnsignedInteger reducedBasisSize = reducedBasis.getSize();
+  Indices allIndices(reducedBasisSize);
+  allIndices.fill();
+  const MatrixImplementation designMatrix(designProxy.computeDesign(allIndices));
+  // The method name is set to the default one, given by ResourceMap
+  String methodName = ResourceMap::GetAsString("LeastSquaresExpansion-DecompositionMethod");
```

Comment:
const + use constructor instead of affectation for the initialization of all the complex types (String or any OT class). Here it should be:
```
const String methodName(ResourceMap::GetAsString("LeastSquaresExpansion-DecompositionMethod"));
```

Reply:
Fixed at d679ef328.